packages/plugins/src/foundry/utils/index.ts

```diff
@@ -89,40 +123,116 @@ export const makeGetConfigArtifacts = (
         time: fs.statSync(`${buildInfoPath}/${fileName}`).mtime.getTime(),
       }))
       .sort((a, b) => b.time - a.time)
-      .map(async (file) => {
-        return JSON.parse(
-          await readFileAsync(join(buildInfoFolder, file.name), 'utf8')
-        )
-      })
 
-    const buildInfos = await Promise.all(buildInfoPromises)
+    // Read all of the new/modified files and update the cache to reflect the changes
+    // Keep an in memory cache of the read files so we don't have to read them again later
+    const localBuildInfoCache = {}
+    await Promise.all(
+      buildInfoFileNames
+        .filter((file) => buildInfoCache[file.name]?.time !== file.time)
+        .map(async (file) => {
+          const buildInfo = JSON.parse(
+            await readFileAsync(join(buildInfoFolder, file.name), 'utf8')
+          )
+
+          // We keep track of the last modified time in each build info file so we can check the file if it has changed.
```

Comment:
I'm pretty sure that each build info file's name is a hash of its contents, so we don't need to worry about a build info file having the same name but different contents. Is there a place in this function where we actually check if the contents have changed? If so, is it necessary? Regardless, it makes sense that we update the file.time